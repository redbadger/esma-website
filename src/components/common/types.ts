export type Fluid = {
  originalName: string;
  base64: string;
  sizes: string;
  src: string;
  srcSet: string;
  aspectRatio: number;
};

export type ChildImageSharp = {
  fluid: Fluid;
};

export type Node = {
  childImageSharp: ChildImageSharp;
};

export type ImageNode = {
  node: Node;
};
