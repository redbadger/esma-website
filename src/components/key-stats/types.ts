import { Stat as PieStat } from "../pie";

export enum KeyStatsCategories {
  Individual = "individual",
  Society = "society",
  Businesses = "businesses",
}

export type ArticleProps = {
  statClasses: (x: string) => string;
};

/// PieProps
/// position: x, y,
/// denotes where on our 300 * 591 map of the UK to put the pie chart
export type PieMapOverlayProps = {
  position: { x: number; y: number };
  pieStats: PieStat[];
  name: string;
};

export type LegendProps = {
  label: string;
  colour: string;
}[];
