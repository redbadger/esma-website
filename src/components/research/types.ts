export type PillData = {
  name: string;
  href: string;
};

export type PillsProps = {
  pills: Array<PillData>;
  colorActive: string;
  currentPillIndex: number;
};

export type ResearchPageMetaData = {
  order: number;
  parent: string;
  title: string;
  color: string;
  bgImageName: string;
  footnotes: FootNoteData[];
};

export type FootNoteData = {
  id: string;
  text: string;
  destination: string;
};

