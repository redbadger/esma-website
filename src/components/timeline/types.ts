export type TimelineEntryData = {
  title: string;
  imageName: string;
  description: string;
  destination: string;
  icon: React.ElementType;
  anchor: string;
};

export type TimelineLegendData = {
  title: string;
  icon: React.ElementType;
  color: string;
  anchor: string;
};
