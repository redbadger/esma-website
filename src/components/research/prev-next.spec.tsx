import React from "react";
import renderer from "react-test-renderer";
import PrevNext from "./prev-next";
import { PillData } from "./types";
import EarlyDevIcon from "../../svg/blocks.svg";

describe("PrevNext", () => {
  const samplePills: PillData[] = [
    {
      name: "Life stage overview",
      href: "/research/early-years/executive-summary",
    },
    {
      name: "Family environment",
      href: "/research/early-years/family-environment",
    },
    {
      name: "Housing and the neighbourhood",
      href: "/research/early-years/housing-and-neighbourhood",
    },
    {
      name: "Early years provision",
      href: "/research/early-years/early-years-provision",
    },
    {
      name: "Impact of Covid-19",
      href: "/research/early-years/impact-of-covid-19",
    },
  ];

  const colorActive = "var(--aqua)";

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PrevNext
          pills={samplePills}
          colorActive={colorActive}
          currentPillIndex={0}
          icon={EarlyDevIcon}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
