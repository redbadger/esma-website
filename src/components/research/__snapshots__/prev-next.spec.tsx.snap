// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevNext renders correctly 1`] = `
<nav
  className="css-98r2c1-arrowHoverCss-prevNextCss-highlightColorCss-PrevNext"
>
  <div
    aria-hidden="true"
    className="spacer"
  />
  <a
    className="prev-next-box css-g9z6el-PrevNextLink"
    href="/research/early-years/family-environment"
  >
    <h4>
      <span
        className="label"
      >
        Next Up
      </span>
      <span
        className="arrow css-84es3n-arrowCss"
      />
    </h4>
    <p>
      <span
        className="icon"
      >
        <test-file-stub />
      </span>
      <span
        className="page-title"
      >
        Family environment
      </span>
    </p>
    <div
      className="mobile-arrow"
    >
      <span
        className="arrow css-84es3n-arrowCss"
      />
    </div>
  </a>
</nav>
`;
